### PR TITLE
CI: workflow file for testing cv_workflow

### PR DESCRIPTION
**Pull request by sugetha24**
## Change Summary

Workflow for running molecule test for `cv_workflow` plugin.

## Component(s) name

`arista.avd.cv_workflow`

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [ ] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by sugetha24** [comment]
NOTE: This can be merged only after merging #3500

File: .github/workflows/new-cvp-integration.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    name: Run molecule test for new cvp integration
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        avd_scenario:
+          - 'deploy_to_cv'
+        ansible_version:
+          - 'ansible-core<2.17.0 --upgrade'
+        # Also test minimum ansible version for one scenario.
```
Comment (ClausHolbechArista) [inline .github/workflows/new-cvp-integration.yml]: Please remove this if not needed.

File: .github/workflows/new-cvp-integration.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  # ----------------------------------- #
+  # Test Requirements
+  # ----------------------------------- #
+  python_requirements:
```
Comment (ClausHolbechArista) [inline .github/workflows/new-cvp-integration.yml]: We should not need this in a dispatched workflow.

File: .github/workflows/new-cvp-integration.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  # DEPLOY_TO_CV MOLECULE
+  # ----------------------------------- #
+  molecule_deploy_to_cv:
+    name: Run molecule test for new cvp integration
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        avd_scenario:
+          - 'deploy_to_cv'
```
Comment (ClausHolbechArista) [inline .github/workflows/new-cvp-integration.yml]:
```suggestion
          - 'cv_workflow'
```

File: .github/workflows/new-cvp-integration.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    # if: needs.file-changes.outputs.config_gen == 'true'
+    steps:
+      - name: 'Set environment variables'
+        run: |
+          echo "PY_COLORS=1" >> $GITHUB_ENV
+          echo "ANSIBLE_FORCE_COLOR=1" >> $GITHUB_ENV
+      - uses: actions/checkout@v3
+      - name: Run molecule action
+        env:
+          CVAAS_AAWG_CI: ${{ secrets.CVAAS_AAWG_CI }}
```
Comment (ClausHolbechArista) [inline .github/workflows/new-cvp-integration.yml:31]: This secret is being printed many times throughout the run, so we cannot merge this until that has been addressed.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: .github/workflows/new-cvp-integration.yml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        with:
+          python-version: ${{ matrix.python_version }}
+      - name: 'Install Python requirements'
+        run: |
+          pip install -r ansible_collections/arista/avd/requirements.txt -r ansible_collections/arista/avd/requirements-dev.txt --upgrade
+
+  # ----------------------------------- #
+  # DEPLOY_TO_CV MOLECULE
+  # ----------------------------------- #
+  molecule_deploy_to_cv:
```
Comment (ClausHolbechArista) [inline .github/workflows/new-cvp-integration.yml]: update name here and in the heading above

File: .github/workflows/new-cvp-integration.yml (hunk 4 above)
Comment (sugetha24) [reply .github/workflows/new-cvp-integration.yml:31]: Fixed with `no_log` in molecule test

**Review by ClausHolbechArista — APPROVED** [review]
LGTM. Thanks

File: .github/workflows/new-cvp-integration.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
+---
+name: "CV integration testing"
+
+# on: pull_request
+on: workflow_dispatch
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
```
Comment (ClausHolbechArista) [inline .github/workflows/new-cvp-integration.yml]: We should cancel in-progress on this one, as it could leave the cloudvision server in some weird state.

File: .github/workflows/new-cvp-integration.yml [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        avd_scenario:
+          - 'cv_workflow'
+        ansible_version:
+          - 'ansible-core<2.17.0 --upgrade'
+    steps:
+      - name: 'Set environment variables'
+        run: |
+          echo "PY_COLORS=1" >> $GITHUB_ENV
+          echo "ANSIBLE_FORCE_COLOR=1" >> $GITHUB_ENV
+      - uses: actions/checkout@v3
```
Comment (gmuloc) [inline .github/workflows/new-cvp-integration.yml]: There should be a newer version out
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/new-cvp-integration.yml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        avd_scenario:
+          - 'cv_deploy'
+        ansible_version:
+          - 'ansible-core<2.17.0 --upgrade'
+    steps:
+      - name: 'Set environment variables'
+        run: |
+          echo "PY_COLORS=1" >> $GITHUB_ENV
+          echo "ANSIBLE_FORCE_COLOR=1" >> $GITHUB_ENV
+      - uses: actions/checkout@v3
```
Comment (gmuloc) [inline .github/workflows/new-cvp-integration.yml]:
```suggestion
      - uses: actions/checkout@v4
```

**Review by gmuloc — APPROVED** [review]
